base/sequenced_task_runner.cc

```diff
@@ -3,12 +3,48 @@
 // found in the LICENSE file.
 
 #include "base/sequenced_task_runner.h"
+#include "base/message_loop/message_loop.h"
+#include "base/synchronization/waitable_event.h"
+#include "base/trace_event/trace_event.h"
 
 #include <utility>
 
 #include "base/bind.h"
 
 namespace base {
+#if defined(STARBOARD)
+namespace {
+
+// Runs the given task, and then signals the given WaitableEvent.
+void RunAndSignal(const base::Closure& task, base::WaitableEvent* event) {
+  TRACE_EVENT0("task", "RunAndSignal");
+  task.Run();
+  event->Signal();
+}
+}  // namespace
+
+void SequencedTaskRunner::PostBlockingTask(const base::Location& from_here,
```

Comment:
This one is only used in web_socket_impl_test.cc. Probably can get rid of this by modifying the test in the followup cl.